src/ImageSharp/Formats/Pbm/BufferedReadStreamExtensions.cs

```diff
@@ -28,7 +28,7 @@ public static void SkipWhitespaceAndComments(this BufferedReadStream stream)
                 {
                     innerValue = stream.ReadByte();
                 }
-                while (innerValue != 0x0a);
+                while (innerValue is not 0x0a and not -0x1);
```

Comment:
Should a similar check be added also in line 52 ?

Reply:
```C#
int current = stream.ReadByte() - 0x30;
if ((uint)current > 9)
{
    break;
}
```
In case of EOF `current == -1 -0x30 == -0x31`. `(uint)-0x31 == SomeHighNum > 9`.

Nevertheless, a test case might be nice, these infinite loops are dangerous. 💀

Reply:
Yeah, that's why I left it. We actually enter that loop in ReadDecimal with this test, but I'll double check before merging.

Reply:
We cover this code with the test so I'm happy. I'll open another PR after this one to backport the fix to 2.X